src/backend_helper.c

```diff
@@ -584,40 +584,55 @@ int get_all_options(PrinterCUPS *p, Option **options)
 
     char **option_names;
     int num_options = get_job_creation_attributes(p, &option_names); /** number of options to be returned**/
-    int i, j;                                                        /**Looping variables **/
-    Option *opts = (Option *)(malloc(sizeof(Option) * num_options)); /**Option array, which will be filled **/
+    int i, j, optsIndex = 0;                                         /**Looping variables **/
+
+    /** Count the number of options that are not be to sent to the frontend.
+        Currently, media-col is the only such option.
+        Others may be found and added later. **/
+    int num_invalid_options = 0;
+    for (i = 0; i < num_options; i++) {
+        if(strcmp(option_names[i], "media-col") == 0)
+            num_invalid_options++;
+    }
+
+    Option *opts = (Option *)(malloc(sizeof(Option) * (num_options-num_invalid_options))); /**Option array, which will be filled **/
     ipp_attribute_t *vals;                                           /** Variable to store the values of the options **/
 
     for (i = 0; i < num_options; i++)
     {
-        opts[i].option_name = option_names[i];
+        if(strcmp(option_names[i], "media-col") == 0)
```

Comment:
... and here.
Do not explicitly mention the blacklisted item on two different points in the program, as if new items get added later you have to add them all the time at these two points. Possible solutions:
1. Define a list/array with all the blacklisted items (currently with one single element), then, at the two points I marked, add innerloops going through the list of items, at the first point to determine the number of bad items, ate the second point to skip the bad items.
2. Simpler: Remove the loop at the fist point I mentioned and always allocate the memory for the full number of items, as only very few will get blacklisted the wasted memory is low.

Reply:
Fixed. Initially, the memory is allocated for all the items. At the final assignment, the memory is truncated if required.